---
title: "The Bacterial and Parasitic Microbiome of Howler Monkeys"
author: "Colin Macfarland"
date: "03/02/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

#setwd("~/Bioinformatics")

#if (!requireNamespace("BiocManager", quietly = TRUE))
  #install.packages("BiocManager")

#BiocManager::install("phyloseq")
#BiocManager::install("DESeq2")
#BiocManager::install("decontam")
#BiocManager::install("ShortRead")

#install.packages("readxl")
#install.packages("tidyverse")

library(phyloseq) #needed moving forward for statistical testing
library(DESeq2) #will be used to do differential abundance
library(decontam) #gets rid of the blanks
library(readxl) #read in the datasheet
library(tidyverse) #needed to help organize the howler metadata
library(ShortRead) #needed to read in the Fasta files
library(ggplot2) #figures
library(ggpubr) #combining figures
library(wesanderson) #colour palette
library(viridis) #colour palette
```

## Parasite Interactions with the Gastrointestinal Bacterial Microbiome

In my first data chapter I asked if a positive helminth infection in the howler monkey gut demonstrates a significant difference in relative abundance of two bacterial orders; Bacteroidales and Clostridiales.  Additionally, I looked at general parasite abundance through morphological identification of parasite eggs, and if the alpha and beta diversity of bacterial communites was significantly different between infected and uninfected howler monkeys.  

Preparing the data for analysis
```{r echo = TRUE, warning = FALSE, message = FALSE}
#import the data and decontaminate the samples
Howler_Metadata16s = read_excel("Howler Metadata.xlsx",
                                sheet = "16s")
asv_fasta_dada = readFasta("dadaspASVs.fa")
asv_tax_dada = read.table("dadaspASVs_taxonomy.txt",
                          header = T,
                          row.names = 1,
                          check.names = F,
                          sep = "\t")
asv_tab_dada = read.table("dadaspASVs_counts.txt",
                          header = T,
                          row.names = 1,
                          check.names = F,
                          sep = "\t")

colnames(asv_tab_dada)
#blanks are last samples, 74 to 84
#logical vector for last 11 samples
decontam_logic_dada = c(rep(TRUE, 73), rep(FALSE, 11))

contam_df_dada = isContaminant(t(asv_tab_dada), neg = decontam_logic_dada)

table(contam_df_dada$contaminant)
contam_asvs_dada = row.names(contam_df_dada[contam_df_dada$contaminant == TRUE,])

# making new count table
asv_tab_no_contam_dada = asv_tab_dada[!row.names(asv_tab_dada) %in% contam_asvs_dada, ]
# making new taxonomy table
asv_tax_no_contam_dada = asv_tax_dada[!row.names(asv_tax_dada) %in% contam_asvs_dada, ]


phyotu16 = otu_table(asv_tab_no_contam_dada, taxa_are_rows = TRUE)
phytax16 = tax_table(as.matrix(asv_tax_no_contam_dada))
phytest16 = phyloseq(phyotu16, phytax16)
phytest16

howmeta.df16 = Howler_Metadata16s %>% select(-sample) %>% as.data.frame
rownames(howmeta.df16) = Howler_Metadata16s$sample
howler.data16 = sample_data(howmeta.df16, errorIfNULL = F)

setdiff(colnames(phyotu16), rownames(howler.data16))
#no differences in names, should merge together 

phytestsamp16 = merge_phyloseq(phytest16, howler.data16)
phytestsamp16

#rename for ease moving forward
phy16sbl = phytestsamp16

#take out blank samples
phy16s = prune_samples(sample_names(phy16sbl)[1:73], phy16sbl)
sample_names(phy16s)
```


I found that some Bacteroidales bacteria were proportionally lower in abundance when helminths were found in the gut, while Clostridiales were proportionally higher in abundance. A similar relationship was found in human and mice experiments in a study by [Ramanan et. al. 2016](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4905769/).  

```{r message = FALSE, warning = FALSE}
helminf = subset_samples(phy16s, Helminth_Infection != "NA")
helminfhighcounts = prune_samples(sample_sums(helminf) > 500, helminf) 
#all were over 1000 so no low read counts for any sample

deseqhelm = phyloseq_to_deseq2(helminfhighcounts, ~ Helminth_Infection)
gm_mean = function(x, na.rm=TRUE){
  exp(sum(log(x[x > 0]), na.rm=na.rm) / length(x))
}
geoMeans = apply(counts(deseqhelm), 1, gm_mean)
deseqhelm = estimateSizeFactors(deseqhelm, geoMeans = geoMeans)
deseqhelm = DESeq(deseqhelm, test="Wald", fitType="local")

res = results(deseqhelm)
res = res[order(res$padj, na.last=NA), ]
alpha = 0.01
sigtab = res[(res$padj < alpha), ]
sigtab = cbind(as(sigtab, "data.frame"),
               as(tax_table(helminfhighcounts)[rownames(sigtab), ], "matrix"))
head(sigtab)

# Phylum order - order from highest to lowest log2fold change for visual appeal
x = tapply(sigtab$log2FoldChange, sigtab$Order, function(x) max(x))
x = sort(x, TRUE)
sigtab$Phylum = factor(as.character(sigtab$Order), levels=names(x))

# Make the figure
p_DiffExp_helm <- ggplot(sigtab, aes(x=Family, y=log2FoldChange, color=Order, size = baseMean)) + 
  geom_point() + 
  geom_hline(yintercept = 0, linetype="dotted", alpha=0.5) +
  theme(axis.text.x = element_text(angle = -50, hjust = 0, vjust=0.5)) +
  scale_colour_viridis(discrete = T); p_DiffExp_helm

allASV = na.omit(res)
allASV = cbind(as(allASV, "data.frame"),
               as(tax_table(helminfhighcounts)[rownames(allASV), ], "matrix"))

p_DiffExp_helm_ALL = ggplot(allASV, aes(x=Class, y=log2FoldChange, color=Phylum)) +
  geom_jitter(size = 1, width = 0.2) +
  theme(axis.text.x = element_text(angle = -90)) +
  scale_colour_viridis(discrete = T); p_DiffExp_helm_ALL

```

Figure **A** showing log fold changes of all ASVs that were classified, and figure **B** showing only those that had a significant log fold change when comparing helminth positive and negative howlers.
```{r echo = FALSE}
figure4combo = ggarrange(p_DiffExp_helm_ALL, p_DiffExp_helm, nrow = 2, labels = "AUTO"); figure4combo
```

## General Parasite Prevalence

When looking at the parasite prevalence, only helminths that could be morphologically identified through microscopy were considered. Smaller parasites such as *Giardia* or *Cryptosporidia* could not be reliably identified without staining or genetic methods.  The percent prevalence is a simple count of the proportion of samples at least one of each parasite was found in.  I found that *Controrchis* spp. was the most prevalent parasite, similar to what has been reported previously, and likely is due to a common plant and staple food source, *Cecropia* spp. 

```{r message = FALSE, warning = FALSE}
paraprev <- read_excel("~/Desktop/Masters /Writing/Figure.xlsx", 
                       sheet = "fig1")
View(paraprev)

Parasite_Prevalence = paraprev %>%
  arrange(Prevalence) %>%  
  mutate(Parasite = factor(Parasite, levels=c("C. biliophilus", "Unidentified Trematode", "Strongyloides sp.", 
                              "Pinworm (Enterobius sp.)", "Unknown"))) %>%
  ggplot(aes(x = Parasite, y = Prevalence)) + 
      geom_col() +
      theme(axis.text.x = element_text(angle = 45, hjust = 1)) + 
      geom_text(aes(label=Prevalence), vjust = -0.4);Parasite_Prevalence
```


## Alpha and Beta Diversity

I did not find significant changes in the alpha and beta diversity of howlers when parasite presence was the tested variable.

```{r warning = FALSE}
library(viridis)
alphtest = estimate_richness(helminf)
modifylab = c("Negative (n = 6)", "Positive (n = 15)")

parasitealpha = plot_richness(helminf, 
              x="Helminth_Infection", 
              measures=c("Chao1","Shannon"), 
              color = "Helminth_Infection") + 
  geom_point() +
  theme_bw() +
  stat_summary(fun = median, fun.min = median, fun.max = median, 
               geom = "crossbar", width = 0.7) +
  theme(legend.position = "none",
        axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1)) +
  labs(x = "Helminth Infection") +
  scale_x_discrete(labels = modifylab) +
  scale_color_manual(values = wes_palette("Darjeeling2"))

#Shannon is diversity. Sig diff in diversity of bacteria, taking into account the richness and evenness
pairwise.wilcox.test(alphtest$Shannon, sample_data(helminf)$Helminth_Infection)
#0.62, no sig diff due to helminth infection
#CHAO1 is richness. Sig diff in richness of bacteria, none
#Simpson measures what shannon does but places less emphasis on rare species, they will have less impact
pairwise.wilcox.test(alphtest$Chao1, sample_data(helminf)$Helminth_Infection)
#0.85, no sig diff due to helm infection

#16s beta diversity clustering
library(microbiome)
phy16hellog = transform(helminf, transform = "log10")
phy16helhell = transform(helminf, transform = "hellinger")

ord.nmds.bray16hel = ordinate(phy16hellog, method="NMDS", distance="bray")
plot_ordination(phy16hellog, ord.nmds.bray16hel, color="Helminth_Infection", title="16s Bray NMDS Infection log trans")

ord.nmds.bray16hel = ordinate(phy16hellog, method="PCoA", distance="bray")
plot_ordination(phy16hellog, ord.nmds.bray16hel, color="Helminth_Infection") +
  scale_color_manual(values = wes_palette("Darjeeling2")) +
  labs(color = "Helminth Infection")


library(vegan)
phydist = phyloseq::distance(phy16hellog, "bray")
df = as(sample_data(phy16hellog), "data.frame")
groups = df[["Helminth_Infection"]]
groups

mod = betadisper(phydist, groups)
anova(mod)
mod.HSD = TukeyHSD(mod)

plot(mod)
boxplot(mod)
plot(mod.HSD)

mod2 = betadisper(phydist, groups, bias.adjust = T)
anova(mod2)
mod.HSD2 = TukeyHSD(mod2)
plot(mod.HSD2)

adonis(phydist ~ groups, perm = 999)

df

library(pairwiseAdonis)
pairwise.adonis(phydist, groups)

ord.nmds.bray16hel2 = ordinate(phy16helhell, method="NMDS", distance="bray")
plot_ordination(phy16helhell, ord.nmds.bray16hel2, color="Helminth_Infection", title="16s Bray NMDS Infection hellinger trans")
```

## Anthopogenic Forest Fragmentation effects on the GI Bacterial Microbiome

In my second chapter I examined if habitat fragmentation/proximity to human development plays a role in lowering bacterial diversity in the howler gastrointestinal microbiome. This was done by looking at 21 howler fecal samples frozen in liquid nitrogen, taken from three locations; Sector Santa Rosa (SSR), located within the Area Conservacion de Guanacaste, a world hertiage site, a social group of howlers near Tamarindo, and a social group of howlers living around a Ranch/Tourist location (Eladio's).

I examined alpha diversity and found that in the howlers living around Eladio's had significantly lowered diversity compared to SSR and Tamarindo howler sample groups.  

```{r warning = FALSE}
#alpha diversity 

library(ggplot2)

#CHAO1 is RICHNESS and SHANNON is DIVERSITY
Costasamp = subset_samples(phy16s, field_site != "Monkey River") #all Costa Rica samples, frozen and ethanol

#alpha significance
alphatest = estimate_richness(Costasamp)

pairwise.wilcox.test(alphatest$Shannon, sample_data(Costasamp)$field_site) 
pairwise.wilcox.test(alphatest$Simpson, sample_data(Costasamp)$field_site)
pairwise.wilcox.test(alphatest$Chao1, sample_data(Costasamp)$field_site)
pairwise.wilcox.test(alphatest$Chao1, sample_data(Costasamp)$fragmentation)
pairwise.wilcox.test(alphatest$Shannon, sample_data(Costasamp)$fragmentation)
#ssr and tam 0.03, ssr and eladio 0.0005, ssr and tam 0.6
#CHAO1 is richness. Sig diff in richness of bacteria, so significance
#Simpson measures similar to shannon, but places less emphasis on rare asvs, they will have less impact

#visualize without untransformed for alpha diversity
library(viridis) #palatte
library(wesanderson) #palatte

#final figures in original thesis, using all the samples together, frozen and ethanol
sample_data(Costasamp)$field_site = factor(sample_data(Costasamp)$field_site, levels = c("Santa Rosa","Eladio's Ranch","Tamarindo"))
Allhabitat = plot_richness(Costasamp, 
              x="field_site", 
              color = "field_site",
              measures=c("Chao1","Shannon", "Simpson")) + 
  theme(legend.title = element_blank()) +
  geom_boxplot() +
  labs(x = "Habitat") +
  scale_color_manual(values = wes_palette("Darjeeling2"))

Fragm = plot_richness(Costasamp,
              x="fragmentation",
              color = "fragmentation",
              measures = c("Chao1", "Shannon", "Simpson")) +
  theme(legend.title = element_blank()) +
  geom_boxplot() +
  labs(x = "Habitat") +
  scale_color_manual(values = wes_palette("Darjeeling2"))
library(ggpubr)
figurealphacombo = ggarrange(Allhabitat, Fragm, nrow = 2, labels = "AUTO")

#Revisions to figures, including only using the frozen samples in the analysis
Frozsamp = subset_samples(phy16s, storage=="Liquid Nitrogen") #only frozen samples
ethsamp = subset_samples(phy16s, storage=="Ethanol")

alphatest = estimate_richness(Frozsamp)
alphatesteth = estimate_richness(ethsamp)
#frozen
pairwise.wilcox.test(alphatest$Shannon, sample_data(Frozsamp)$field_site) 
pairwise.wilcox.test(alphatest$Simpson, sample_data(Frozsamp)$field_site)
pairwise.wilcox.test(alphatest$Chao1, sample_data(Frozsamp)$field_site)
pairwise.wilcox.test(alphatest$Chao1, sample_data(Frozsamp)$fragmentation)
pairwise.wilcox.test(alphatest$Shannon, sample_data(Frozsamp)$fragmentation)

#more correct method for analysis
#Santa vs Eladios
alpharesults = estimate_richness(Frozsamp, measures = "Chao1")
d = sample_data(Frozsamp)
A = alpharesults[d[, 'field_site'] == "Santa Rosa",]
B = alpharesults[d[, 'field_site'] == "Eladio's Ranch",]
wilcox.test(A, B)
#p-value = 0.02373

alpharesults2 = estimate_richness(Frozsamp, measures = "Simpson")
d2 = sample_data(Frozsamp)
A2 = alpharesults[d[, 'field_site'] == "Santa Rosa",]
B2 = alpharesults[d[, 'field_site'] == "Eladio's Ranch",]
wilcox.test(A2, B2)
#p-value = 0.02373

alpharesults3 = estimate_richness(Frozsamp, measures = "Shannon")
d3 = sample_data(Frozsamp)
A3 = alpharesults[d[, 'field_site'] == "Santa Rosa",]
B3 = alpharesults[d[, 'field_site'] == "Tamarindo",]
wilcox.test(A3, B3)

alpharesults4 = estimate_richness(Frozsamp, measures = "Shannon")
d4 = sample_data(Frozsamp)
A4 = alpharesults[d[, 'field_site'] == "Eladio's Ranch",]
B4 = alpharesults[d[, 'field_site'] == "Tamarindo",]
wilcox.test(A4, B4)

#santa and fragmented
alpharesults5 = estimate_richness(Frozsamp, measures = "Shannon")
d5 = sample_data(Frozsamp)
A5 = alpharesults[d[, 'fragmentation'] == "Santa Rosa",]
B5 = alpharesults[d[, 'fragmentation'] == "Fragmented",]
wilcox.test(A5, B5)

alpharesults6 = estimate_richness(Frozsamp, measures = "Chao1")
d6 = sample_data(Frozsamp)
A6 = alpharesults[d[, 'fragmentation'] == "Santa Rosa",]
B6 = alpharesults[d[, 'fragmentation'] == "Fragmented",]
wilcox.test(A6, B6)

#ethanol
pairwise.wilcox.test(alphatesteth$Shannon, sample_data(ethsamp)$field_site) 
pairwise.wilcox.test(alphatesteth$Simpson, sample_data(ethsamp)$field_site)
pairwise.wilcox.test(alphatesteth$Chao1, sample_data(ethsamp)$field_site)
pairwise.wilcox.test(alphatesteth$Chao1, sample_data(ethsamp)$fragmentation)
pairwise.wilcox.test(alphatesteth$Shannon, sample_data(ethsamp)$fragmentation)

#frozen results
#Field Shannon - ssr/elad - 0.07, ssr/tam - 0.686, elad/tam - 0.686
#Field Simpson - ssr/elad - 0.094, ssr/tam - 0.517, elad/tam - 0.4
#Field Chao1 - ssr/elad - 0.24, ssr/tam - 1, elad/tam - 1
#Frag Chao1 - ssr/frag - 0.11
#Field Shannon - srr/frag - 0.075

#with all tests, an alpha of 0.05, no significance in bacterial alpha diversity among only frozen samples (n=21 rather than n=42 in original).  With the ethanol samples, Santa Rosa and Eladio's are significantly different in diversity measueres, but not richness.

#visualize without untransformed for alpha diversity
library(viridis) #palatte
library(wesanderson) #palatte
install.packages("ggsignif")
library(ggsignif)

#final figures frozen, dotplots instead of boxplots, and median line added
sample_data(Frozsamp)$field_site = factor(sample_data(Frozsamp)$field_site, levels = c("Santa Rosa","Eladio's Ranch","Tamarindo"))
Allhabitat = plot_richness(Frozsamp, 
              x="field_site",
              color = "field_site",
              measures=c("Chao1","Shannon", "Simpson")) + 
  theme_bw() +
  theme(legend.title = element_blank(), legend.position = "none",
        axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1)) +
  geom_point() +
  geom_signif(comparisons = list(c("Santa Rosa","Eladio's Ranch")), 
              map_signif_level = c("***" = 0.001, "**" = 0.01, "*" = 0.05)) +
  stat_summary(fun = median, fun.min = median, fun.max = median, 
               geom = "crossbar", width = 0.7) +
  labs(x = "Habitat") +
  scale_color_manual(values = wes_palette("Darjeeling2"))

sample_data(Frozsamp)$fragmentation = factor(sample_data(Frozsamp)$fragmentation, levels = c("Santa Rosa","Fragmented"))
Fragm = plot_richness(Frozsamp,
              x="fragmentation",
              color = "fragmentation",
              measures = c("Chao1", "Shannon", "Simpson")) +
  theme_bw() +
  theme(legend.title = element_blank(), legend.position = "none",
        axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1)) +
  geom_point() +
  stat_summary(fun = median, fun.min = median, fun.max = median, 
               geom = "crossbar", width = 0.7) +
  labs(x = "Habitat") +
  scale_color_manual(values = wes_palette("Darjeeling2"))

#final figures ethanol
sample_data(ethsamp)$field_site = factor(sample_data(ethsamp)$field_site, levels = c("Santa Rosa","Eladio's Ranch","Tamarindo"))
Allhabitateth = plot_richness(ethsamp, 
              x="field_site",
              color = "field_site",
              measures=c("Chao1","Shannon", "Simpson")) + 
  theme_bw() +
  theme(legend.title = element_blank(), legend.position = "none",
        axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1)) +
  geom_point() +
  geom_signif(comparisons = list(c("Santa Rosa","Eladio's Ranch")), 
              map_signif_level = c("***" = 0.001, "**" = 0.01, "*" = 0.05)) +
  stat_summary(fun = median, fun.min = median, fun.max = median, 
               geom = "crossbar", width = 0.7) +
  labs(x = "Habitat") +
  scale_color_manual(values = wes_palette("Darjeeling2"))

sample_data(ethsamp)$fragmentation = factor(sample_data(ethsamp)$fragmentation, levels = c("Santa Rosa", "Fragmented"))
Fragmeth = plot_richness(ethsamp,
              x="fragmentation",
              color = "fragmentation",
              measures = c("Chao1", "Shannon", "Simpson")) +
  theme_bw() +
  theme(legend.title = element_blank(), legend.position = "none",
        axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1)) +
  geom_point() +
  stat_summary(fun = median, fun.min = median, fun.max = median, 
               geom = "crossbar", width = 0.7) +
  labs(x = "Habitat") +
  scale_color_manual(values = wes_palette("Darjeeling2"))

#combining figures; frozen and ethanol side by side, Habitat then Fragmentation
library(ggpubr)
figurealphacombohab = ggarrange(Allhabitat, Allhabitateth, nrow = 2, labels = "AUTO"); figurealphacombohab

figurealphacombofrag = ggarrange(Fragm, Fragmeth, nrow = 2, labels = "AUTO"); figurealphacombofrag
```

I additionally found that the howler beta diversity was distinct between all three sites, an ADONIS test showing that howlers in SSR, Tamarindo, and Eladio's had distinct bacterial diversity in their gut.   

```{r warning = FALSE}
#beta diversity
#use transformed data
#betadisper to check for homogeneous dispersal and acceptable variance

library(vegan)
#install.packages("devtools")
library(devtools)
#install_github("pmartinezarbizu/pairwiseAdonis/pairwiseAdonis")
library(pairwiseAdonis)

library(microbiome)
phy16log = transform(Costasamp, transform = "log10")

phydist = phyloseq::distance(phy16log, "bray")
df = as(sample_data(phy16log), "data.frame")
groups = df[["field_site"]]
groups

mod = betadisper(phydist, groups)
anova(mod)
mod.HSD = TukeyHSD(mod)

plot(mod)
boxplot(mod)
plot(mod.HSD)

mod2 = betadisper(phydist, groups, bias.adjust = T)
anova(mod2)
mod.HSD2 = TukeyHSD(mod2)
plot(mod.HSD2)

adonis(phydist ~ groups, perm = 999)

df

pairwise.adonis(phydist, groups)
#betadisper testing shows acceptable variance and homogeneous dispersal, can continue to adonis, and results can be interpreted
#pairwise adonis between all groups, showing all have significantly different bacterial diversity

#other transformation options
#phy16hell = transform(Costasamp, transform = "hellinger")
#phy16sprop2 = transform_sample_counts(Costasamp, function(otu) otu/sum(otu))

ord.nmds.bray16log = ordinate(phy16log, method="PCoA", distance="bray")
plot_ordination(phy16log, ord.nmds.bray16log, color="field_site") +
  scale_color_manual(values = wes_palette("Darjeeling2")) +
  labs(color = "Habitat") +
  stat_ellipse()
#PCoA plot of Bray Curtis dissimilarity, with ellipses to show clustering of groups, determined to be significant with the pairwise ADONIS

#Revisions to figure including only using frozen samples in analysis
library(microbiome)
phy16logfr = transform(Frozsamp, transform = "log10")

phydist = phyloseq::distance(phy16logfr, "bray")
df = as(sample_data(phy16logfr), "data.frame")
groups = df[["field_site"]]
groups

mod = betadisper(phydist, groups)
anova(mod)
mod.HSD = TukeyHSD(mod)

plot(mod)
boxplot(mod)
plot(mod.HSD)

mod2 = betadisper(phydist, groups, bias.adjust = T)
anova(mod2)
mod.HSD2 = TukeyHSD(mod2)
plot(mod.HSD2)

adonis(phydist ~ groups, perm = 999)

df

pairwise.adonis(phydist, groups)
#betadisper testing shows acceptable variance and homogeneous dispersal, can continue to adonis, and results can be interpreted
#pairwise adonis between all groups, showing all have significantly different bacterial diversity

#other transformation options
#phy16hell = transform(Costasamp, transform = "hellinger")
#phy16sprop2 = transform_sample_counts(Costasamp, function(otu) otu/sum(otu))

ord.nmds.bray16log = ordinate(phy16logfr, method="PCoA", distance="bray")
plot_ordination(phy16logfr, ord.nmds.bray16log, color="field_site") +
  scale_color_manual(values = wes_palette("Darjeeling2")) +
  labs(color = "Habitat") +
  stat_ellipse()
#PCoA plot of Bray Curtis dissimilarity, with ellipses to show clustering of groups, determined to not be significant with the pairwise ADONIS 
```

Finally, I found that howlers in SSR, the most isolated group, had the highest density of parasite infection, measured in parasite eggs per mL of fecal sample (after the sample was homogenized in ethanol it was stored in). 

```{r warning = FALSE}
paralocaldens <- read_excel("Desktop/Masters /Writing/Figure.xlsx", 
                            sheet = "fig2edit")
View(paralocaldens)

Parasite_Density = paralocaldens %>%
  arrange(AveDens) %>%  
  mutate(Location = factor(Location, levels=c("Santa Rosa (n = 13)", "Fragmented (n = 8)"))) %>%
  ggplot(aes(x = Location, y = AveDens)) + 
                      geom_col(aes(fill = Location)) +
                      ylab("Density (eggs/mL)") +
  scale_fill_manual(values = wes_palette("Darjeeling2")) +
  theme(legend.position = "none") +
  geom_errorbar(aes(ymin=AveDens-SE, ymax=AveDens+SE), width=.05); Parasite_Density
```

## Additional Testing
Does the storage method, either ethanol or liquid nitrogen have a significant effect on bacterial diversity?

```{r}
#alpha diversity and beta diversity 

library(ggplot2)

#Storage media comparison
Costasamp = subset_samples(phy16s, field_site != "Monkey River") #all Costa Rica samples, frozen and ethanol

#alpha significance
alphatest = estimate_richness(Costasamp)

pairwise.wilcox.test(alphatest$Shannon, sample_data(Costasamp)$storage) 
pairwise.wilcox.test(alphatest$Simpson, sample_data(Costasamp)$storage)
pairwise.wilcox.test(alphatest$Chao1, sample_data(Costasamp)$storage)

#Shannon - p value 0.83 
#Simpson - p value 0.95
#Chao1 - p value 0.0078
#Simpson measures similar to shannon, but places less emphasis on rare asvs, they will have less impact, Chao1 significant, richness of bacterial species is significantly different, diversity measures not significant 

#visualize without untransformed for alpha diversity
library(viridis) #palatte
library(wesanderson) #palatte

#final figures
Storagefig = plot_richness(Costasamp, 
              x="storage", 
              color = "storage",
              measures=c("Chao1","Shannon", "Simpson")) + 
  theme(legend.title = element_blank()) +
  geom_boxplot() +
  labs(x = "Storage Method") +
  scale_color_manual(values = wes_palette("Darjeeling2")); Storagefig

#beta diversity of sample storage comparison
library(vegan)
#install.packages("devtools")
library(devtools)
#install_github("pmartinezarbizu/pairwiseAdonis/pairwiseAdonis")
library(pairwiseAdonis)

BiocManager::install("microbiome")
library(microbiome)
phy16log = transform(Costasamp, transform = "log10")

phydist = phyloseq::distance(phy16log, "bray")
df = as(sample_data(phy16log), "data.frame")
groups = df[["storage"]]
groups

mod = betadisper(phydist, groups)
anova(mod)
mod.HSD = TukeyHSD(mod)

plot(mod)
boxplot(mod)
plot(mod.HSD)

mod2 = betadisper(phydist, groups, bias.adjust = T)
anova(mod2)
mod.HSD2 = TukeyHSD(mod2)
plot(mod.HSD2)

adonis(phydist ~ groups, perm = 999)

df

pairwise.adonis(phydist, groups, p.adjust.m = 'holm')
#betadisper testing shows acceptable variance and homogeneous dispersal, can continue to adonis, and results can be interpreted
#pairwise adonis between all groups, showing all have significantly different bacterial diversity 0.04, but figure showing extreme overlap between groups

ord.nmds.bray16log = ordinate(phy16log, method="PCoA", distance="bray")
plot_ordination(phy16log, ord.nmds.bray16log, color="storage") +
  scale_color_manual(values = wes_palette("Darjeeling2")) +
  labs(color = "Storage Method") +
  stat_ellipse() 
```

